app/controllers/users/sessions_controller.rb

```diff
@@ -79,6 +80,40 @@ def process_locked_out_session
       redirect_to root_url
     end
 
+    def valid_captcha_result?
+      if cookies[:device] && (device = Device.find_by(cookie_uuid: cookies[:device]))
+        return true if device.user.email_addresses.lazy.map(&:email).include?(auth_params[:email])
+      end
+
+      response, _assessment_id = recaptcha_form.submit(params.require(:user)[:recaptcha_token])
+      flash[:error] = response.first_error_message if !response.success?
+      response.success?
+    end
+
+    def process_failed_captcha
+      warden.logout(:user)
+      warden.lock!
+      redirect_to root_url
+    end
+
+    def recaptcha_form
+      @recaptcha_form ||= SignInRecaptchaForm.new(**recaptcha_form_args)
+    end
+
+    def recaptcha_form_args
+      args = { analytics: }
+      if IdentityConfig.store.phone_recaptcha_mock_validator
+        args.merge(
+          form_class: RecaptchaMockForm,
+          score: params.require(:user)[:recaptcha_mock_score].to_f,
+        )
+      elsif FeatureManagement.recaptcha_enterprise?
+        args.merge(form_class: RecaptchaEnterpriseForm)
+      else
+        args
```

Comment:
should we default to `RecaptchaForm` for completeness?
```suggestion
        args.merge(form_class: RecaptchaForm)
```

Reply:
It's defaulted in `SignInRecaptchaForm`, so not strictly necessary:

https://github.com/18F/identity-idp/blob/2dd7c9ce98d3a92e9e1c6eace4c0ae09529d5a85/app/forms/sign_in_recaptcha_form.rb#L10

This follows from the implementation for phone setup:

https://github.com/18F/identity-idp/blob/37afb7efb468486efe3f7ed50ad7600f6329fc37/app/forms/new_phone_form.rb#L144-L153

But I could also change it so that `form_class` is a required keyword attribute and assign it here instead.

(Aside: When implementing this "proper", I'll probably plan to create a separate form class for handling the sign-in+reCAPTCHA validation, similar to what we have with `NewPhoneForm`)

Reply:
yeah seeing the default argument there is what prompted me to make the comment here